tests/platform_tests/api/test_sfp.py

```diff
@@ -689,8 +689,15 @@ def test_lpmode(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, pla
                 logger.warning("test_lpmode: Skipping transceiver {} (not applicable for this transceiver type)".format(i))
                 continue
 
+            lpmode_state_pretest = sfp.get_lpmode(platform_api_conn, i)
+            if lpmode_state_pretest is None:
+                logger.warning("test_lpmode: Skipping transceiver {} (not supported on this platform)".format(i))
+                break
+            # This order makes sure lpmode will get restored to pretest value after test
+            lpmode_states_to_be_tested = [not lpmode_state_pretest, lpmode_state_pretest]
```

Comment:
@longhuan-cisco can we use explicit [True, False] and not depend upon the read value of pretest?

Reply:
the reasons of doing this are: 
1. changing the order of lpmode_states_to_be_tested sequence according to value of lpmode_pretest will make sure lpmode state gets restored to the pre-test state at the end of test. e.g. 1) if lpmode_pretest is True, then lpmode_states_to_be_tested is [False, True], which will set lpmode back to True in the end. 2) if lpmode_pretest is False, then lpmode_states_to_be_tested is [True, False], which will set lpmode back to False in the end.
2. Also, if we blindly set lpmode_states_to_be_tested to [True, False], then if lpmode_pretest is True, then in the first round of testing, it will set lpmode to True although it was already True, there's not much point of doing that. 
3. By doing this, we don't need to explicitly restore the lpmode state at the end (since it was already restored), which saves one lpmode set action.